builtin/credential/aws/cli.go

```diff
@@ -99,7 +99,11 @@ func (h *CLIHandler) Auth(c *api.Client, m map[string]string) (*api.Secret, erro
 		return nil, err
 	}
 
-	loginData, err := GenerateLoginData(creds, headerValue, m["region"])
+	region := m["region"]
+	if region == "" {
+		region = awsutil.DefaultRegion
+	}
+	loginData, err := GenerateLoginData(creds, headerValue, region)
```

Comment:
Shouldn't `GenerateLoginData` call `GetOrDefaultRegion` which defaults to us-east-1 if `m["region"]` is not provided and we're not able to determine the region via other means?

Reply:
It's the "determine the region via other means" that has caused a number of issues since 1.2. This PR is effectively reverting that behavior.